src/bluesearch/database/article.py

```diff
@@ -227,24 +215,78 @@ def get_paragraphs_sections_mapping(self) -> Dict[Element, str]:
 
         return mapping
 
-    @staticmethod
-    def text_content(element: Optional[Element]) -> str:
-        """Extract all text of an element and of its descendants (at any depth).
+    def _inner_text(self, element: Element) -> str:
+        """Convert all inner text and sub-elements to one string.
+
+        In short, we collect all the inner text while also converting all
+        sub-elements that we encounter to strings using ``self._element_to_str``.
+        All escaped HTML in the raw text is unescaped.
+
+        For example, if schematically the element is given by
+
+            element = "<p>I <bold>like</bold> python &amp; ice cream.<p>"
+
+        then ``_inner_text(element)`` would give
+
+            "I like python & ice cream."
+
+        provided that "<bold>like</bold>" is resolved to "like" by the
+        ``self._element_to_str`` method.
+
+        Parameters
+        ----------
+        element
+            The input XML element.
+
+        Returns
+        -------
+        str
+            The inner text and sub-elements converted to one single string.
+        """
+        text_parts = [html.unescape(element.text or "")]
+        for sub_element in element:
+            # recursively parse the sub-element
+            text_parts.append(self._element_to_str(sub_element))
+            # don't forget the text after the sub-element
+            text_parts.append(html.unescape(sub_element.tail or ""))
+        return "".join(text_parts).strip()
+
+    def _element_to_str(self, element: Element | None) -> str:
+        """Convert an element and all its contents to a string.
 
         Parameters
         ----------
-        element : etree._Element or None
-            XML element to parse.
+        element
+            The input XML element.
 
         Returns
         -------
         str
-            Entire text of the element and its descendants.
+            A parsed string representation of the input XML element.
         """
         if element is None:
             return ""
+
+        if element.tag in {"p", "bold", "italic", "underline", "monospace", "xref"}:
```

Comment:
I think `sc` and `styled-content` should be fine. But for `disp-quote` it's usually long-ish block quotes from patients etc. (from `PMC3426202.nxml`, re-formatted):
```xml
                    <p>Others had received helpful safer conception information from doctors or
                        nurses at PMTCT, ARV, and gynecology clinics. Risk reduction strategies that
                        participants had learned included artificial insemination, intercourse timed
                        to peak fertility, manual insemination (for male&#x02014;uninfected couples&#x02014;the
                        uninfected male ejaculates into a condom or other container and the semen is
                        inserted into the woman's vaginal canal via a syringe or reversed condom),
                        sperm washing, and intercourse with lubrication (to avoid abrasions).
                        <disp-quote>
                            <p>They told me that I can go to one of those private hospitals to ask
                                about sperm washing&#x02026; One of the times when I came into the
                                clinic there was a poster on the wall saying &#x0201c;if you want to
                                have a child, speak to either a doctor or a nurse.&#x0201d; So I
                                asked the doctor, &#x0201c;do you think is it possible for me to
                                have a child even if I'm positive?&#x0201d; But I had to go to one
                                of those private hospitals to look into that thing [sperm washing].
                                <italic>(Participant B05, 31-year-old man)</italic>
                            </p>
                        </disp-quote>
                        <disp-quote>
                            <p>A doctor gave us a [syringe] which we had to use after sex to
                                withdraw sperm [from the condom to insert into the woman]. &#x02026;
                                We are both aware that I am positive and so we would use a condom
                                all the time &#x02026; and we received advice from the doctor on how
                                to use the syringe that he gave us so that he [partner] does not get
                                infected.
                                <italic>(Participant A25, 32-year-old woman)</italic>
                            </p>
                        </disp-quote>
```
Maybe we can be slightly more careful here:
- They seem to almost be paragraphs on their own
- Should we wrap them into quotes?
- Them being paragraph-like there aren't always space/newline characters around so that after the default extraction the first/last words might get glued together.

Maybe a separate issue for this?

Reply:
Seems reasonable. We can add just `sc` and `styled-content` for the moment, and deal with `disp-quote` in another issue.

Reply:
Added `<sc>` and `<styled-content>` here: 592aaf9

Reply:
Created #450